modules/markup/html.go

```diff
@@ -1054,6 +1060,266 @@ func comparePatternProcessor(ctx *RenderContext, node *html.Node) {
+		if err != nil {
+			return
+		}
+
+		status := &charset.EscapeStatus{}
+		statuses := make([]*charset.EscapeStatus, len(fileContent))
+		for i, line := range fileContent {
+			statuses[i], fileContent[i] = charset.EscapeControlHTML(line, locale)
+			status = status.Or(statuses[i])
+		}
+
+		for idx, code := range fileContent {
+			tr := &html.Node{
+				Type: html.ElementNode,
+				Data: atom.Tr.String(),
+			}
+
+			lineNum := strconv.Itoa(lineOffset + idx + 1)
+
+			tdLinesnum := &html.Node{
+				Type: html.ElementNode,
+				Data: atom.Td.String(),
+				Attr: []html.Attribute{
+					{Key: "id", Val: "L" + lineNum},
+					{Key: "class", Val: "lines-num"},
+				},
+			}
+			spanLinesNum := &html.Node{
+				Type: html.ElementNode,
+				Data: atom.Span.String(),
+				Attr: []html.Attribute{
+					{Key: "id", Val: "L" + lineNum},
+					{Key: "data-line-number", Val: lineNum},
+				},
+			}
+			tdLinesnum.AppendChild(spanLinesNum)
+			tr.AppendChild(tdLinesnum)
+
+			if status.Escaped {
+				tdLinesEscape := &html.Node{
+					Type: html.ElementNode,
+					Data: atom.Td.String(),
+					Attr: []html.Attribute{
+						{Key: "class", Val: "lines-escape"},
+					},
+				}
+
+				if statuses[idx].Escaped {
+					btnTitle := ""
+					if statuses[idx].HasInvisible {
+						btnTitle += locale.Tr("repo.invisible_runes_line") + " "
+					}
+					if statuses[idx].HasAmbiguous {
+						btnTitle += locale.Tr("repo.ambiguous_runes_line")
+					}
+
+					escapeBtn := &html.Node{
+						Type: html.ElementNode,
+						Data: atom.A.String(),
+						Attr: []html.Attribute{
+							{Key: "class", Val: "toggle-escape-button btn interact-bg"},
+							{Key: "title", Val: btnTitle},
+							{Key: "href", Val: "javascript:void(0)"},
+						},
+					}
+					tdLinesEscape.AppendChild(escapeBtn)
+				}
+
+				tr.AppendChild(tdLinesEscape)
+			}
+
+			tdCode := &html.Node{
+				Type: html.ElementNode,
+				Data: atom.Td.String(),
+				Attr: []html.Attribute{
+					{Key: "rel", Val: "L" + lineNum},
+					{Key: "class", Val: "lines-code chroma"},
+				},
+			}
+			codeInner := &html.Node{
+				Type: html.ElementNode,
+				Data: atom.Code.String(),
+				Attr: []html.Attribute{{Key: "class", Val: "code-inner"}},
+			}
+			codeText := &html.Node{
+				Type: html.RawNode,
+				Data: string(code),
+			}
+			codeInner.AppendChild(codeText)
+			tdCode.AppendChild(codeInner)
+			tr.AppendChild(tdCode)
+
+			tbody.AppendChild(tr)
+		}
+
+		table.AppendChild(tbody)
+
+		twrapper := &html.Node{
+			Type: html.ElementNode,
+			Data: atom.Div.String(),
+			Attr: []html.Attribute{{Key: "class", Val: "ui table"}},
+		}
+		twrapper.AppendChild(table)
+
+		header := &html.Node{
+			Type: html.ElementNode,
+			Data: atom.Div.String(),
+			Attr: []html.Attribute{{Key: "class", Val: "header"}},
+		}
+		afilepath := &html.Node{
+			Type: html.ElementNode,
+			Data: atom.A.String(),
+			Attr: []html.Attribute{{Key: "href", Val: urlFull}},
+		}
```

Comment:
Add a `muted` class to this to uncolor it.

Reply:
Should this too be applied for the second link to the commit itself?

Rendering with only muted on the first link:

![grafik](https://github.com/go-gitea/gitea/assets/67418776/1e2b118a-e392-425f-8e78-0d3abff06bdf)

Reply:
Yes, but in this case I would say add `text grey` classes to the parent element. and `text black` classes to the link so that it stands out a bit.